src/helpers/http.js

```diff
@@ -47,12 +60,21 @@ function resolveHttpPromise(httpEvent, resolveFunction, startTime) {
  * @param {object} httpEvent The current event
  * @param {string} key name in metadata
  * @param {string} data data to jsonify
+ * @param {string} encoding data encoding from the headers
  */
-function setJsonPayload(httpEvent, key, data) {
+function setJsonPayload(httpEvent, key, data, encoding) {
     try {
-        JSON.parse(data);
+        let jsonData = data;
+        if (DECODE_HTTP && ENCODING_FUNCTIONS[encoding]) {
+            try {
+                jsonData = ENCODING_FUNCTIONS[encoding](data);
+            } catch (err) {
+                utils.debugLog(`Could not parse ${encoding} ${key} in http`);
+            }
+        }
+        JSON.parse(jsonData);
         eventInterface.addToMetadata(httpEvent, {}, {
-            [key]: data,
+            [key]: jsonData.toString(),
```

Comment:
why to invoke `toString()` in encoding block above?

Reply:
because in case it is a not encoded data, the `data` field is type `Buffer` which needs to be converted to string. Same applies for the decoding functions

Reply:
but then the `JSON.parse()` above still works ?
wouldn't it be better to call `jsonData.toString('utf-8')` ?